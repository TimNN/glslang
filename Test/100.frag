// okay
#version 100
int a[3] = { 2, 3, 4, };  // ERROR

int uint;

attribute vec4 v[3];     // ERROR

float f = 2;             // ERROR

uniform block {          // ERROR
    int x;
};

void foo(float);

void main()
{
    foo(3);              // ERROR
    int s = 1 << 4;      // ERROR
    s = 16 >> 2;         // ERROR
    if (a == a);         // ERROR
    int b, c;
    b = c & 4;           // ERROR
    b = c % 4;           // ERROR
    b = c | 4;           // ERROR
    b >>= 2;             // ERROR
    b <<= 2;             // ERROR
    b %= 3;              // ERROR

    struct S {
        float f;
        float a[10];
    } s1, s2;

    s1 = s2;             // ERROR
    if (s1 == s2);       // ERROR
    if (s1 != s2);       // ERROR

    switch(b) {          // ERROR
    }
}

invariant gl_FragColor;
float fa[];              // ERROR
float f13;
invariant f13;           // ERROR
struct S { int a; };
invariant S;
invariant float fi;      // ERROR
varying vec4 av;
invariant av;            // okay in v100

void foo10()
{
    invariant f;         // ERROR
    invariant float f2;  // ERROR
    float f3;
    invariant f3;        // ERROR
}

uniform vec2 uv2;
invariant uv2;              // ERROR
invariant uniform vec3 uv3; // ERROR

sampler2D glob2D;           // ERROR
void f11(sampler2D p2d)
{
    sampler2D v2D;          // ERROR
}
varying sampler2D vary2D;   // ERROR

struct sp {
    highp float f;
    in float g;             // ERROR
    uniform float h;        // ERROR
    invariant float i;      // ERROR
};
