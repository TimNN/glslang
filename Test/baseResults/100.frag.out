ERROR: 0:3: '{ } style initializers' : not supported with this profile: es
ERROR: 0:3: 'initializer' : not supported for this version or the enabled extensions 
ERROR: 0:7: 'attribute' : not supported in this stage: fragment
ERROR: 0:7: 'float' : type requires declaration of default precision qualifier 
ERROR: 0:9: '=' :  cannot convert from 'const int' to 'mediump float'
ERROR: 0:11: 'uniform block' : not supported for this version or the enabled extensions 
ERROR: 0:19: 'foo' : no matching overloaded function found 
ERROR: 0:20: 'bit shift left' : not supported for this version or the enabled extensions 
ERROR: 0:21: 'bit shift right' : not supported for this version or the enabled extensions 
ERROR: 0:22: 'array comparison' : not supported for this version or the enabled extensions 
ERROR: 0:24: 'bitwise and' : not supported for this version or the enabled extensions 
ERROR: 0:25: '%' : not supported for this version or the enabled extensions 
ERROR: 0:26: 'bitwise inclusive or' : not supported for this version or the enabled extensions 
ERROR: 0:27: 'bit-shift right assign' : not supported for this version or the enabled extensions 
ERROR: 0:28: 'bit-shift left assign' : not supported for this version or the enabled extensions 
ERROR: 0:29: '%=' : not supported for this version or the enabled extensions 
ERROR: 0:36: 'array assignment' : not supported for this version or the enabled extensions 
ERROR: 0:37: 'array comparison' : not supported for this version or the enabled extensions 
ERROR: 0:38: 'array comparison' : not supported for this version or the enabled extensions 
ERROR: 0:40: 'switch' : Reserved word. 
ERROR: 0:40: 'switch statements' : not supported for this version or the enabled extensions 
ERROR: 0:45: '' : array size required 
ERROR: 0:47: 'invariant' : can only apply to an output or an input in a non-vertex stage
 
ERROR: 0:50: 'invariant' : can only apply to an output or an input in a non-vertex stage
 
ERROR: 0:56: 'invariant' : can only apply to an output or an input in a non-vertex stage
 
ERROR: 0:57: 'invariant' : can only apply to an output or an input in a non-vertex stage
 
ERROR: 0:59: 'invariant' : can only apply to an output or an input in a non-vertex stage
 
ERROR: 0:63: 'invariant' : can only apply to an output or an input in a non-vertex stage
 
ERROR: 0:64: 'invariant' : can only apply to an output or an input in a non-vertex stage
 
ERROR: 0:66: 'sampler2D' : sampler/image types can only be used in uniform variables or function parameters: glob2D
ERROR: 0:69: 'sampler2D' : sampler/image types can only be used in uniform variables or function parameters: v2D
ERROR: 0:71: 'sampler2D' : sampler/image types can only be used in uniform variables or function parameters: vary2D
ERROR: 0:75: 'g' : cannot use storage or interpolation qualifiers on structure members 
ERROR: 0:76: 'h' : cannot use storage or interpolation qualifiers on structure members 
ERROR: 0:77: 'i' : cannot use invariant qualifier on structure members 
ERROR: 0:80: 'sampler3D' : Reserved word. 
ERROR: 0:80: 'sampler/image' : type requires declaration of default precision qualifier 
ERROR: 37 compilation errors.  No code generated.

ERROR: node is still EOpNull!
0:3  Sequence
0:3    move second child to first child (3-element array of mediump int)
0:3      'a' (3-element array of mediump int)
0:3      Constant:
0:3        2 (const int)
0:3        3 (const int)
0:3        4 (const int)
0:17  Function Definition: main( (void)
0:17    Function Parameters: 
0:19    Sequence
0:19      Constant:
0:19        0.000000
0:20      Sequence
0:20        move second child to first child (mediump int)
0:20          's' (mediump int)
0:20          Constant:
0:20            16 (const int)
0:21      move second child to first child (mediump int)
0:21        's' (mediump int)
0:21        Constant:
0:21          4 (const int)
0:22      Test condition and select (void)
0:22        Condition
0:22        Compare Equal (bool)
0:22          'a' (3-element array of mediump int)
0:22          'a' (3-element array of mediump int)
0:22        true case is null
0:24      move second child to first child (mediump int)
0:24        'b' (mediump int)
0:24        bitwise and (mediump int)
0:24          'c' (mediump int)
0:24          Constant:
0:24            4 (const int)
0:25      move second child to first child (mediump int)
0:25        'b' (mediump int)
0:25        mod (mediump int)
0:25          'c' (mediump int)
0:25          Constant:
0:25            4 (const int)
0:26      move second child to first child (mediump int)
0:26        'b' (mediump int)
0:26        inclusive-or (mediump int)
0:26          'c' (mediump int)
0:26          Constant:
0:26            4 (const int)
0:27      right shift second child into first child (mediump int)
0:27        'b' (mediump int)
0:27        Constant:
0:27          2 (const int)
0:28      left shift second child into first child (mediump int)
0:28        'b' (mediump int)
0:28        Constant:
0:28          2 (const int)
0:29      mod second child into first child (mediump int)
0:29        'b' (mediump int)
0:29        Constant:
0:29          3 (const int)
0:36      move second child to first child (structure{f,a})
0:36        's1' (structure{f,a})
0:36        's2' (structure{f,a})
0:37      Test condition and select (void)
0:37        Condition
0:37        Compare Equal (bool)
0:37          's1' (structure{f,a})
0:37          's2' (structure{f,a})
0:37        true case is null
0:38      Test condition and select (void)
0:38        Condition
0:38        Compare Not Equal (bool)
0:38          's1' (structure{f,a})
0:38          's2' (structure{f,a})
0:38        true case is null
0:40      'b' (mediump int)
0:54  Function Definition: foo10( (void)
0:54    Function Parameters: 
0:67  Function Definition: f11(s21; (void)
0:67    Function Parameters: 
0:67      'p2d' (in lowp sampler2D)
0:87  Function Definition: foo234( (void)
0:87    Function Parameters: 
0:89    Sequence
0:89      Function Call: texture3D(s31;vf3;f1; (highp 4-component vector of float)
0:89        's3D2' (uniform highp sampler3D)
0:89        Constant:
0:89          0.200000
0:89          0.200000
0:89          0.200000
0:89        Constant:
0:89          0.200000
0:90      Function Call: texture3DProj(s31;vf4;f1; (highp 4-component vector of float)
0:90        's3D2' (uniform highp sampler3D)
0:90        direct index (smooth mediump 4-component vector of float)
0:90          'v' (smooth in 3-element array of mediump 4-component vector of float)
0:90          Constant:
0:90            1 (const int)
0:90        Constant:
0:90          0.400000
0:?   Linker Objects
0:?     'a' (3-element array of mediump int)
0:?     'uint' (mediump int)
0:?     'v' (smooth in 3-element array of mediump 4-component vector of float)
0:?     'f' (mediump float)
0:?     '__anon__0' (layout(column_major shared ) uniform block{x})
0:?     'fa' (unsized array of mediump float)
0:?     'f13' (mediump float)
0:?     'fi' (invariant mediump float)
0:?     'av' (smooth in mediump 4-component vector of float)
0:?     'uv2' (uniform mediump 2-component vector of float)
0:?     'uv3' (invariant uniform mediump 3-component vector of float)
0:?     'glob2D' (lowp sampler2D)
0:?     'vary2D' (smooth in lowp sampler2D)
0:?     's3D' (uniform mediump sampler3D)
0:?     's3D2' (uniform highp sampler3D)

